### ISSUE TITLE
Add Object Detection, Image Segmentation & Model Visualization

### ISSUE THREAD

**Issue by 5usu** [SRA-VJTI/computer-vision-study-group#6]
## Summary
Extend the repo to include **object detection**, **semantic segmentation**, and **model interpretability** tools.  

---

##  Objective
- Use pretrained networks for detection & segmentation
- Implement Grad-CAM and feature visualization for interpretability
- Work with OpenCV + PyTorch to visualize outputs

---

##  Tasks
- [ ] Create folder: `detection_segmentation/`
- [ ] Implement:
  - `detect.py` — object detection using pretrained YOLO/SSD models
  - `segment.py` — semantic segmentation (DeepLabv3 / FCN)
  - `gradcam.py` — Grad-CAM visualization on CNN models (from Issue #2)
  - `utils.py` — preprocessing, color maps, result visualization
- [ ] Add sample scripts for webcam/image inference
- [ ] Add visualization outputs (bounding boxes, masks, Grad-CAM heatmaps)

---

##  Directory Structure
cv_learn/
detection_segmentation/
detect.py
segment.py
gradcam.py
utils.py
README.md

---

## Tools / Dependencies
- Python 3.10+
- PyTorch + torchvision
- OpenCV
- matplotlib
- numpy
- tqdm

---

##  Example CLI
```bash
# Object Detection
python detection_segmentation/detect.py --model yolov8n --image sample.jpg --out results/detect.jpg

# Image Segmentation
python detection_segmentation/segment.py --model deeplabv3_resnet50 --image sample.jpg --out results/seg.jpg

# Grad-CAM Visualization
python detection_segmentation/gradcam.py --weights ../cnn_classification/model_cnn.pth --image sample.jpg --layer conv3

**Comment by badboy1606** [comment]
Hello maintainers, can you please assign this enhancement to me , would ensure proper implementation and good contribution.
